src/libcore/sync/atomic.rs

```diff
@@ -1391,6 +1482,52 @@ macro_rules! atomic_int {
             pub fn fetch_xor(&self, val: $int_type, order: Ordering) -> $int_type {
                 unsafe { atomic_xor(self.v.get(), val, order) }
             }
+
+            /// Maximum with the current value.
+            ///
+            /// Finds the maximum of the current value and the argument `val`, and sets the new
+            /// value to the result.
+            ///
+            /// Returns the maximum value.
```

Comment:
This sentence is wrong, it returns the *previous* value. This is strictly more useful but examples should show how to recover the newly stored maximum or minimum:

```rust
if x.fetch_max(y, SeqCst) < y { y } else { x }
```

Reply:
so that would be `x.fetch_max(y, SeqCst).max(y)`, right?

Reply:
I guess, yes.